install_dependencies.sh

```diff
@@ -1,30 +1,99 @@
 #!/bin/bash
 
 
+help_install_dependencies="Usage:
+./install_dependencies.sh [os_name]         default OS is linux
+./install_dependencies.sh --help            displays help
+
+os_name can be:
+  - linux
+  - osx
+
+it installes the given dependencies:
+  - odb compiler                            odb-2.4.0
+  - odb common runtime library              libodb-2.4.0
+  - odb sqlite runtime library              libodb-sqlite-2.4.0
+
+for linux:
+./install_dependencies.sh linux
+
+for mac os:
+./install_dependencies.sh osx
+"
+
+if [ "$#" -eq 0 ]
+then
+    OS_NAME="linux"
+elif [ "$#" -eq 1 ]
+then
+  if [[ "$1" == "--help" ]]
+  then
+      OS_NAME="$help_install_dependencies"
+  elif [[ "$1" == "linux" ]]
+  then
+      OS_NAME="linux"
+  elif [[ "$1" == "osx" ]]
+  then
+      OS_NAME="osx"
+  else
+    echo "$help_install_dependencies"
+    exit
+  fi
+else
+  echo "requires 1 argument at max"
+  echo "$help_install_dependencies"
+  exit
+fi
+
+odb=$OS_NAME"_dependencies"
+
+#check for already downloaded files
+if [ -d "$odb/libodb-2.4.0" ]; then
+  echo "skipping odb installation: $odb directory found:"
+  ls $odb
+  exit
+fi
+
+
 # download and compile ODB compiler, ODB runtime library, and sqlite DB plugin for ODB
-if [ ! -d "odb/odb-2.4.0-x86_64-linux-gnu" ]; then
-  mkdir -p odb && cd odb
-  wget http://codesynthesis.com/download/odb/2.4/odb-2.4.0-x86_64-linux-gnu.tar.bz2 -O /tmp/odb.tar.bz2
-  tar jxvf /tmp/odb.tar.bz2
-  # the odb compiler is already available as binary: no configure && make
+echo creating directory $odb
```

Comment:
Quotes for consistency with rest of messages.

Reply:
hmm I havent yet focused much on messages. right now my focus was on the structure. I will update it .